### PR TITLE
Fix Android support by bypassing SELinux

### PR DESCRIPTION
**Pull request by MeWnoJinsei**
Use /data directory and /system/build.prop to identify android device, since `/proc/sys/kernel/ostype` is inaccessible on later version of Android (>=5.0) because of SELinux policies.

### CODE REVIEW

**Comment by 0x0ade** [comment]
Does the directory `/etc/selinux` exist on your device? I can only check with Termux right now, and *something* is linking `/etc` to `/system/etc`, dunno if it's Android or Termux.

If it does, I'd like to change the check so that:
- `/etc/selinux` implies Linux,
- `/etc/selinux` + the two Android-specific paths imply Android,
- anything else falls through to `/proc/sys/kernel/ostype` and then the macOS `SystemVersion` check.

This should also fix possible issues in other Linux environments with SELinux enforced.

**Comment by MeWnoJinsei** [comment]
> Does the directory `/etc/selinux` exist on your device? I can only check with Termux right now, and _something_ is linking `/etc` to `/system/etc`, dunno if it's Android or Termux.
> 
> If it does, I'd like to change the check so that:
> 
>     * `/etc/selinux` implies Linux,
> 
>     * `/etc/selinux` + the two Android-specific paths imply Android,
> 
>     * anything else falls through to `/proc/sys/kernel/ostype` and then the macOS `SystemVersion` check.
> 
> 
> This should also fix possible issues in other Linux environments with SELinux enforced.

Yes it does, and the symbol link /etc -> /system/etc is documented in [here](https://source.android.com/devices/bootloader/partitions/system-as-root#about-system-as-root), giving us a steady proof of doing that.

File: Utils/PlatformHelper.cs [hunk 1]
```diff
@@ -61,13 +64,13 @@ static class PlatformHelper {
                 _current = Platform.Android;
 
             } else if (Is(Platform.Unix) &&
-                Directory.Exists("/Applications") && Directory.Exists("/System") &&
-                Directory.Exists("/User") && !Directory.Exists("/Users")
+              Directory.Exists("/Applications") && Directory.Exists("/System") &&
+              Directory.Exists("/User") && !Directory.Exists("/Users")
```
Comment (0x0ade) [inline Utils/PlatformHelper.cs:68]: Can you please revert this change?

File: Utils/PlatformHelper.cs [hunk 2]
```diff
@@ -61,13 +64,13 @@ static class PlatformHelper {
...
             ) {
                 _current = Platform.iOS;
 
             } else if (Is(Platform.Windows) &&
-                CheckWine()
+              CheckWine()
```
Comment (0x0ade) [inline Utils/PlatformHelper.cs:73]: Can you please revert this change?

File: Utils/PlatformHelper.cs [hunk 3]
```diff
@@ -21,6 +21,9 @@ static class PlatformHelper {
             if (!string.IsNullOrEmpty(windir) && windir.Contains(@"\", StringComparison.Ordinal) && Directory.Exists(windir)) {
                 _current = Platform.Windows;
 
+            } else if (Directory.Exists("/etc/selinux")){   // SELinux enabled, access to /proc/sys/kernel/ostype might fail
+                _current = Platform.Linux;  // We cannot directly use /proc/sys/kernel/ostype to determine an Android Device
+                                              // because of the strict SELinux policies after Lollipop
```
Comment (0x0ade) [inline Utils/PlatformHelper.cs:26]:
```suggestion
            } else if (Directory.Exists("/etc/selinux")) {
                // SELinux enabled, access to /proc/sys/kernel/ostype might fail
                // We cannot directly use /proc/sys/kernel/ostype to determine an Android Device
                // because of the strict SELinux policies after Lollipop
                _current = Platform.Linux;
```

**Review by 0x0ade — CHANGES_REQUESTED** [review]
Just a few minor code style complaints, otherwise LGTM 👍 thank you :)

**Comment by 0x0ade** [comment]
I've just merged your changes with a subsequent cleanup commit. Thank you :)
